### ISSUE TITLE
LanguageManager - custome error codes not working with translations

### ISSUE THREAD

**Issue by wniar** [JeremySkinner/FluentValidation#926]
Hi,

I implement custom version of ILanguageManager with two translations: 

```
AddTranslation("en", "NotNullValidator", "Can't be empty");
AddTranslation("en", "EmailTest", "Invalid email");
```

Second of my translation is my custom validator:

```
public static IRuleBuilderOptions<T, string> CustomEmailAddress<T>(this IRuleBuilder<T, string> ruleBuilder)
{
    return ruleBuilder.Must(x => EmailValidator.Validate(x)).WithErrorCode("EmailTest");
}
```

When my custom email error trigger i not get message from my translation. I got message from previous code ("PredicateValidator"). I test this and when i add translation for code which exists they translate correct. This look like message generate before error code is changed or changing error code not affect to resource property name. @JeremySkinner can you say this is expected behavior?

**Comment by JeremySkinner** [comment]
Hi there, yes this is expected behaviour. I think you've misunderstood how the LanguageManager works.

The LanguageManager provides a way to override the default error messages for validators. It is based on the validator type and has nothing to do with error codes. The use of "PredicateValidator" is not the error code here - it is the validator's type name.

For example, a call to `Must` internally uses the `PredicateValidator` class. If you wanted to replace the default message for all calls to `Must` in your code then you would use `AddTranslation("en", "PredicateValidator", "foo")`. 

You cannot use the LanguageManager in the way you're trying (based on error code). For specific uses of `Must`, you should be using `WithMessage` to provide the default message inside your extension method. 

If you really want to do this, you would need to create your own class that inherits from `PredicateValidator`, and then use that type name in your language manager.

Hope that makes sense.

**Comment by JeremySkinner** [comment]
I do actually like the idea of being able to do this though, so it’s something that can be looked at for a future version.

**Comment by wniar** [comment]
Thanks for your answer. That make sense, now i understand your idea. 

This will be great if all translations can be defined in one place.

**Comment by JeremySkinner** [comment]
I've implemented this in 8.1.0-preview2 which is now available on nuget. Please give it a try and let me know if you run into any issues.

**Comment by wniar** [comment]
@JeremySkinner I made tests. Everything work as i expect, great job and great future!
